app/models/manager_refresh/inventory_collection/builder.rb

```diff
@@ -0,0 +1,252 @@
+module ManagerRefresh
+  class InventoryCollection
+    # @see test in /spec/models/manager_refresh/inventory_collection/builder_spec.rb
+    class Builder
+      class MissingModelClassError < StandardError; end
+
+      require_nested :CloudManager
+      require_nested :InfraManager
+      require_nested :AutomationManager
+      require_nested :NetworkManager
+      require_nested :StorageManager
+      require_nested :PersisterHelper
+
+      include ::ManagerRefresh::InventoryCollection::Builder::Shared
+
+      # Default options for builder
+      #   :adv_settings
+      #     - values from Advanced settings (doesn't overwrite values specified in code)
+      #     - @see method ManagerRefresh::Inventory::Persister.options()
+      #   :auto_model_class
+      #     - tries to set model_class from persister class
+      #     - @see method auto_model_class
+      #   :auto_inventory_attributes
+      #     - auto creates inventory_object_attributes from target model_class setters
+      #     - attributes used in InventoryObject.add_attributes
+      #   :shared_properties
+      #     - any properties applied if missing (not explicitly specified)
+      #   :without_model_class
+      #     - if false and no model_class derived or specified, throws exception
+      def self.default_options
+        {
+          :adv_settings              => {},
+          :auto_model_class          => true,
+          :auto_inventory_attributes => true,
```

Comment:
I have to admit I'm not a fan of extra configuration to turn these things on/off I think it adds complexity.
We should leave them always on or always off, and if they don't work for some reason (probably because of a weird nuance with that particular IC) then that collection should override the model_class/inventory_attributes.

@Ladas @slemrmartin thoughts?  I think the extra options makes this more complex.

Reply:
I don't know how to do it - "override model_class/inventory_attributes", can you show me some example?

And there are some unusual ICs:
See this IC: https://github.com/ManageIQ/manageiq-providers-amazon/pull/438/files#diff-0f974d25c81f9b0090224fd98112a9d6R106

There is no model class and no inventory attributes (if you skip this options, tests will fail) - how to solve these kind of collections?

Reply:
> I don't know how to do it - "override model_class/inventory_attributes", can you show me some example?

I.e. make them manually specify the model_class, and only auto-guess the model class if one isn't provided by the caller.

Reply:
then from my point of view, if one "guessing" can be enabled/disabled, then all "guessings" should have same option, it seems more consistent to me

Reply:
One more consideration - auto_model_class can be merged with without_model_class. 
The second option should skip auto-guessing too. If we want some special class, we have to define it manually, if there will be no class, auto-guessing has to be disabled (because there cannot be guessed some existing class (or future existing class) accidentally). I believe there isn't other case

Reply:
I mean remove it as an option from `default_options`, the more things that can be tweaked and turned on/off the more complex it is.
I think we should _always_ auto-model-class and if it doesn't work we should override it.

As for auto-inventory-attributes, I haven't started looking at that method yet but I agree that is probably the sketchiest and maybe we go back to explicit inventory-attributes for now and tackle the auto in a standalone PR.

Reply:
about ```auto_model_class``` - I really don't like removing it from documented part of options and make it as hidden settings, that's worse, I think. I join ```auto_model_class``` and ```without_model_class``` together, as I wrote.

about ```auto_inventory_attributes``` - I refactored 6 providers and all tests passes...I really don't want to get rid of it and rewrite it all including this PR just because splitting work to 2 PRs only.

Reply:
> I really don't like removing it from documented part of options and make it as hidden settings

I don't think it would be a hidden setting, it would be always on.  And we should still document it but we can document it in the prepare_data method.

This isn't something that we're going to want to ever turn off in the field, we'll know if we can guess the model class correctly or not so I don't see a point to having it as a setting it is just increasing complexity.